src/main/scala/edu/ie3/mobsim/model/ElectricVehicle.scala

```diff
@@ -46,21 +46,18 @@ case class ElectricVehicle(
     private val consumption: ComparableQuantity[SpecificEnergy], // kWh per km
     private val homePoi: PointOfInterest,
     private val workPoi: PointOfInterest,
-    private var storedEnergy: ComparableQuantity[Energy],
-    private var destinationPoiType: PoiTypeDictionary.Value,
-    private var destinationCategoricalLocation: CategoricalLocationDictionary.Value,
-    private var destinationPoi: PointOfInterest,
-    private var parkingTimeStart: ZonedDateTime,
-    private var departureTime: ZonedDateTime,
+    private val storedEnergy: ComparableQuantity[Energy],
+    private val destinationPoi: PointOfInterest,
+    private val parkingTimeStart: ZonedDateTime,
+    private val departureTime: ZonedDateTime,
```

Comment:
I really like this so far. Using `val` over `var` is already a big improvement.
Using some properties of Scala case classes, we can push things even further: The getters below are obsolete if we remove the `private` modifiers here. Using case classes, there is no need for getters anymore, since all fields are immutable now. You can even leave out the `val` modifier, since all fields of a case class are immutable per default.

Reply:
Dome getters are still needed because this class ist an implementation of the ``EvModel`` found in the Simona API.